api/v1alpha1/imageclusterinstall_types.go

```diff
@@ -34,14 +34,36 @@ const (
 	HostConfigurationSucceededMessage = "Configuration image is attached to the referenced host"
 )
 
-// ClusterConfigSpec defines the desired state of ClusterConfig
-type ClusterConfigSpec struct {
-	clusterinfo.ClusterInfo `json:",inline"`
+// ImageClusterInstallSpec defines the desired state of ImageClusterInstall
+type ImageClusterInstallSpec struct {
+	// ClusterDeploymentRef is a reference to the ClusterDeployment.
+	// +optional
+	ClusterDeploymentRef *corev1.LocalObjectReference `json:"clusterDeploymentRef"`
+
+	// ImageSetRef is a reference to a ClusterImageSet.
+	ImageSetRef hivev1.ClusterImageSetReference `json:"imageSetRef"`
+
+	// ClusterMetadata contains metadata information about the installed cluster.
+	// This must be set as soon as all the information is available.
+	// +optional
+	ClusterMetadata *hivev1.ClusterMetadata `json:"clusterMetadata"`
+
+	// Version is the target OCP version for the cluster
+	// TODO: should this use ImageSetRef?
+	Version string `json:"version,omitempty"`
+
+	// MasterIP is the desired IP for the host
+	MasterIP string `json:"masterIP,omitempty"`
 
-	// PullSecretRef is a reference to new cluster-wide pull secret.
-	// If defined, it will replace the secret located at openshift-config/pull-secret.
-	// The type of the secret must be kubernetes.io/dockerconfigjson.
-	PullSecretRef *corev1.LocalObjectReference `json:"pullSecretRef,omitempty"`
+	// Hostname is the desired hostname for the host
+	Hostname string `json:"hostname,omitempty"`
+
+	// ReleaseRegistry is the registry of the installed release image
+	// TODO: should this also use the registry of ImageSetRef?
+	ReleaseRegistry string `json:"releaseRegistry,omitempty"`
```

Comment:
@omertuc what exactly is this? Would it have anything to do with the release image?

Reply:
[Baffled](https://github.com/omertuc/lifecycle-agent/commit/e3a82df9d37f6f5e8c08172c8f0246c470c35b0c#diff-e6ccbfc77f9d37a80162c13f8b94d2044699b8d090451a8848d2b4686b6ff815R45) me as well

Reply:
@donpenney , @tsorya does any of you know what this field is and what it's for?

Reply:
This is specifically for the release image reference, which I think is explicitly the mirror. It required special handling in the case of disconnected registries as a result, as it uses the explicit registry reference and not the original quay.io ref like other images. @tsorya did a lot of work to handle these cases, and could likely provide a better explanation

Reply:
So from what I understand, this is required in case the seed image was installed using a release image from a mirror registry that is different from the registry the cluster should use.
We use this field to override the opensfhit-cluster-version deployment `release-image` parameter with an updated release image FQDN that uses the registry the cluster should use. 

Note: this is relevant for cases where the release image is specified with the FQDN from the mirror registry and not redirected via ICSP or IDSP.